examples/linear_model/plot_tweedie_regression_insurance_claims.py

```diff
@@ -0,0 +1,597 @@
+
+df = load_mtpl2(n_samples=60000)
+
+# Note: filter out claims with zero amount, as the severity model
+# requires strictly positive target values.
+df.loc[(df["ClaimAmount"] == 0) & (df["ClaimNb"] >= 1), "ClaimNb"] = 0
+
+# Correct for unreasonable observations (that might be data error)
+# and a few exceptionally large claim amounts
+df["ClaimNb"] = df["ClaimNb"].clip(upper=4)
+df["Exposure"] = df["Exposure"].clip(upper=1)
+df["ClaimAmount"] = df["ClaimAmount"].clip(upper=200000)
+
+log_scale_transformer = make_pipeline(
+    FunctionTransformer(func=np.log),
+    StandardScaler()
+)
+
+column_trans = ColumnTransformer(
+    [
+        ("binned_numeric", KBinsDiscretizer(n_bins=10),
+            ["VehAge", "DrivAge"]),
+        ("onehot_categorical", OneHotEncoder(),
+            ["VehBrand", "VehPower", "VehGas", "Region", "Area"]),
+        ("passthrough_numeric", "passthrough",
+            ["BonusMalus"]),
+        ("log_scaled_numeric", log_scale_transformer,
+            ["Density"]),
+    ],
+    remainder="drop",
+)
+X = column_trans.fit_transform(df)
+
+# Insurances companies are interested in modeling the Pure Premium, that is
+# the expected total claim amount per unit of exposure for each policyholder
+# in their portfolio:
+df["PurePremium"] = df["ClaimAmount"] / df["Exposure"]
+
+# This can be indirectly approximated by a 2-step modeling: the product of the
+# Frequency times the average claim amount per claim:
+df["Frequency"] = df["ClaimNb"] / df["Exposure"]
+df["AvgClaimAmount"] = df["ClaimAmount"] / np.fmax(df["ClaimNb"], 1)
+
+with pd.option_context("display.max_columns", 15):
+    print(df[df.ClaimAmount > 0].head())
+
+##############################################################################
+#
+# Frequency model -- Poisson distribution
+# ---------------------------------------
+#
+# The number of claims (``ClaimNb``) is a positive integer (0 included).
+# Thus, this target can be modelled by a Poisson distribution.
+# It is then assumed to be the number of discrete events occuring with a
+# constant rate in a given time interval (``Exposure``, in units of years).
+# Here we model the frequency ``y = ClaimNb / Exposure``, which is still a
+# (scaled) Poisson distribution, and use ``Exposure`` as `sample_weight`.
+
+df_train, df_test, X_train, X_test = train_test_split(df, X, random_state=0)
+
+# The parameters of the model are estimated by minimizing the Poisson deviance
+# on the training set via a quasi-Newton solver: l-BFGS. Some of the features
+# are collinear, we use a weak penalization to avoid numerical issues.
+glm_freq = PoissonRegressor(alpha=1e-3, max_iter=400)
+glm_freq.fit(X_train, df_train["Frequency"],
+             sample_weight=df_train["Exposure"])
+
+scores = score_estimator(
+    glm_freq,
+    X_train,
+    X_test,
+    df_train,
+    df_test,
+    target="Frequency",
+    weights="Exposure",
+)
+print(scores)
+
+##############################################################################
+#
```

Comment:
```suggestion
```
I don't know why you keep inserting these empty lines at the beginning of sections. It should not be necessary I think.